### PR TITLE
Enable LibCST-based autofixing for SPR001

### PR DESCRIPTION
**Pull request by charliermarsh**
# Summary

This PR demonstrates use of LibCST to autofix a specific lint error (rather than using it as an end-to-end parser solution).

### CODE REVIEW

**Comment by charliermarsh** [comment]
\cc @sobolevn

File: src/autofix/fixes.rs [hunk 1]
```diff
@@ -124,3 +125,34 @@ pub fn remove_class_def_base(
...
+        Ok(m) => m,
+        Err(_) => return None,
+    };
+
+    if let Some(Statement::Simple(body)) = tree.body.first_mut() {
+        if let Some(SmallStatement::Expr(body)) = body.body.first_mut() {
+            if let Expression::Call(body) = &mut body.value {
+                body.args = vec![];
+                body.whitespace_before_args = Default::default();
+                body.whitespace_after_func = Default::default();
```
Comment (charliermarsh) [inline src/autofix/fixes.rs:142]: Here's the actual logic: extract the `Call`, then zero-out the arguments.

File: resources/test/fixtures/SPR001.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def method(self):
 
     def wrong(self):
         parent = super(Child, self)  # wrong
-        super(Child, self).method  # wrong
+        super(Child, self).method()  # wrong
```
Comment (sobolevn) [inline resources/test/fixtures/SPR001.py:18]: This was on purpose: `ast.Attribute` vs `ast.Call`

File: src/snapshots/ruff__linter__tests__spr001.snap [hunk 3]
```diff
@@ -6,14 +6,39 @@ expression: checks
...
     row: 19
     column: 9
-  fix: ~
+  fix:
+    content: super()
+    start:
+      row: 19
+      column: 9
+    end:
+      row: 22
```
Comment (sobolevn) [inline src/snapshots/ruff__linter__tests__spr001.snap:41]: Should not it be `row: 19` now?

Basically, what it is says in the snapshop:

```python
super(


       ).method() # wrong
```

Or, on the other hand, we can leave this to black or similar tools to fix.

**Review by sobolevn — COMMENTED** [review]
Oh, good stuff!

File: resources/test/fixtures/SPR001.py (hunk 2 above)
Comment (charliermarsh) [reply resources/test/fixtures/SPR001.py:18]: Oh right, will revert. (The code works fine w/ either, didn't realize it was intentional but makes total sense.)

File: src/snapshots/ruff__linter__tests__spr001.snap (hunk 3 above)
Comment (charliermarsh) [reply src/snapshots/ruff__linter__tests__spr001.snap:41]: This is the start and end of the lines in the _unfixed_ source code, so I think it's correct?

Comment (sobolevn) [reply src/snapshots/ruff__linter__tests__spr001.snap:41]: Oh, I see! 👍

**Pull request merged**